build.gradle

```diff
@@ -62,6 +63,29 @@ spotless {
 
 jar { dependsOn checkLicenses }
 
+class FileStateChecker extends DefaultTask {
+    Set<File> files
+    String knownHash
+
+    @TaskAction
+    def CheckState() {
+        def allContents = this.files.sort()*.readLines()*.join('\n').join('')
+        def currentHash = MessageDigest.getInstance("SHA-256")
+        .digest(allContents.getBytes("UTF-8")).encodeBase64().toString()
+        if (this.knownHash != currentHash) {
+            throw new GradleException("Was expecting: ${this.knownHash}\nGot instead: ${currentHash}")
```

Comment:
We should make the message explicitly talk about the plugin APIs so people know the context.  Probably should be fairly long winded and specifically state if the changes were deliberate and you've considered backwards compatibility then update the hash to <blah>...